src/Aspire.Hosting.Azure/Provisioning/Internal/BaseProvisioningContextProvider.cs

```diff
@@ -149,6 +149,12 @@ public virtual async Task<ProvisioningContext> CreateProvisioningContextAsync(Js
 
         var principal = await _userPrincipalProvider.GetUserPrincipalAsync(cancellationToken).ConfigureAwait(false);
 
+        // Persist the provisioning options to user secrets so they can be reused in the future
+        var azureSection = userSecrets.Prop("Azure");
```

Comment:
Rename `userSecrets` to `deploymentState` here and elsewhere to make it clear that this migh not exclusively be user secrets.